python/hail/dataset.py

```diff
@@ -5809,4 +5809,9 @@ def make_table(self, variant_expr, genotype_expr, key=[], separator='.'):
                                 jarray(Env.jvm().java.lang.String, wrap_to_list(key)), separator)
         return KeyTable(self.hc, jkt)
 
+    def _to_new_variant_dataset(self):
```

Comment:
OK, I think you should make a second module for the new python expr interface.  It should be self-contained and there should conversion routines between the old and new setups.  Call the new one `newhail` or `hail2` or `hailpyexpr` or something.  (I'll call it `hail2`.)  There should be a `hail2.HailContext` that returns `hail2.KeyTable`, etc.

`hail.VariantDataset` should have `to_hail2` and `hail2.VariantDataset` should similarly have `to_hail1` (or `to_oldhail` or whatever.)